pro/src/pages/IndividualOffer/IndividualOfferDetails/commons/utils.ts

```diff
@@ -97,32 +86,58 @@ export const completeSubcategoryConditionalFields = (
     ...(subcategory?.isEvent ? ['durationMinutes'] : []),
   ] as (keyof DetailsFormValues)[]
 
-export const formatVenuesOptions = (
-  venues: VenueListItemResponseModel[],
-  isOnline: boolean
-) => {
-  //  We want to display the virtual venues only if there are no physical venues available
-  //  We also want to prevent selecting a virtual venue for a physical offer form
+/**
+ * **Rules**
+ * 1. Virtual venues are shown **only when no physical venue exists**.
+ * 2. When this is a **physical** offer (`isOfferVirtual === false`), virtual venues are never shown.
+ */
+export function filterAvailableVenues(
+  venues: readonly VenueListItemResponseModel[],
+  isOfferVirtual: boolean
+): VenueListItemResponseModel[] {
   const hasAtLeastOnePhysicalVenue = venues.some((v) => !v.isVirtual)
+  // Virtual venues are allowed only if *all* venues are virtual *and* the offer is virtual.
+  const shouldIncludeVirtualVenues =
+    !hasAtLeastOnePhysicalVenue && isOfferVirtual
+
+  return venues.filter((v) => shouldIncludeVirtualVenues || !v.isVirtual)
```

Comment:
Au fond est-ce que ça ne serait pas plus simple ainsi  ?
```suggestion
): VenueListItemResponseModel[] {
  const physicalVenues = venues.filter((v) => !v.isVirtual)
  // Virtual venues are allowed only if *all* venues are virtual *and* the offer is virtual.
  if (isOfferVirtual && physicalVenues.length === 0) {
     return venues;
  }
  
  return physicalVenues
```